analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.TryCatchFinally.cs

```diff
@@ -1436,9 +1446,12 @@ public void Exception_MultipleConversion()
                 "BeforeTry",
                 "InTry",
                 "InCatch",
+                "InCatch",
                 "AfterCatch");
 
-            validator.TagStates("InCatch").Should().HaveCount(1).And.ContainSingle(x => HasExceptionOfType(x, "FileNotFoundException"));
+            validator.TagStates("InCatch").Should().HaveCount(2)
+                     .And.ContainSingle(x => HasExceptionOfType(x, "FileNotFoundException"))
+                     .And.ContainSingle(x => HasUnknownException(x)); // ObjectCreation can throw unknown exception
```

Comment:
This comment doesn't sound right. You've already implemented, that exception constructors do not throw.

Isn't it coming from conversion instead?

Can we improve the conversion then? Something like if it's a conversion to a known basetype/interface, than do not throw, because those are safe.

Reply:
Wrong comment. It's from conversion. Proper exception will be asserted in the follow-up PR